NMS-14212: Event docs cleanup (#5002)
OpenNMS/opennms: docs/modules/operation/pages/events/event-debugging.adoc

= Event Troubleshooting

== Debugging

When debugging events, it may be helpful to lower the minimum severity at which `Eventd` will log from the default level of `WARN`.
To change this setting, edit `$\{OPENNMS_HOME}/etc/log4j2.xml` and locate the following line:

[source,xml]
----
<KeyValuePair key="eventd" value="WARN" />
----

> Changes to `log42.xml` will take effect within 60 seconds with no extra action needed.

Reviewer: ```suggestion
Changes to `log42.xml` will take effect within 60 seconds with no further action needed.
```

> At level `DEBUG`, Eventd will log a verbose description of every event it handles to `$\{OPENNMS_HOME}/logs/eventd.log`.

Reviewer: ```suggestion
At the `DEBUG` level, Eventd will log a verbose description of every event it handles to `$\{OPENNMS_HOME}/logs/eventd.log`.
```

On busy systems, this setting may create so much noise as to be impractical.

> In these cases, you can get terse event summaries by setting Eventd to log at level `INFO` and setting `logEventSummaries="yes"` in `$\{OPENNMS_HOME}/etc/eventd-configuration.xml`.

Reviewer 2: ```suggestion
In these cases, you can get terse event summaries by setting eventd to log at level `INFO` and setting `logEventSummaries="yes"` in `$\{OPENNMS_HOME}/etc/eventd-configuration.xml`.
```

[[send-event]]
== Manually send events

> You can manually create and send an event to trigger processes inside of {page-component-title}, through the UI or using the <<event-utility, send-event.pl utility>>.

Reviewer: ```suggestion
You can manually create and send an event to trigger processes inside of {page-component-title} through the UI or using the <<event-utility, send-event.pl utility>>.
```

> Most often, you would do this to reload daemon configuration without having to restart {page-component-title}.

Reviewer: ```suggestion
In most cases, you would do this to reload daemon configuration without having to restart {page-component-title}.
```

[[send-event-ui]]
To manually send an event in the UI, follow these steps:

. In {page-component-title} click the menu:Gears Icon[Event Management > Manually Send an Event].
. Select the event from the *Event* list.

> . Specify additional information, including the UUID, Node ID, parameters, severity, and so on and click *Send Event*.

Reviewer: ```suggestion
. Specify additional information, including the UUID, Node ID, parameters, severity, and so on, and click *Send Event*.
```

* open a ticket in your integrated ticketing system

> * order new ink cartridges when your printer is out of toner

Reviewer: The sentences in this list should be capitalized and have a period at the end, but I can't offer suggestions for half of them through the GitHub diff editor.

Format: `$\{OPENNMS_HOME}/bin/send-event.pl <uei> [host:port] [options]`

> The `host:port` defines the hostname and port to send the event; by default, the local host.

Reviewer: ```suggestion
The `host:port` defines the hostname and port to send the event; by default, it points to the local host.
```

Find UEIs and their associated parameters in the `$\{OPENNMS_HOME}/etc/eventd-configuration.xml` file.

> See also xref:events/event-configuration.adoc#ga-events-event-configuration[Event Daemon Configuration].

Reviewer: ```suggestion
See also xref:events/event-configuration.adoc#ga-events-event-configuration[Event daemon configuration].
```

* open a ticket in your integrated ticketing system
* order new ink cartridges when your printer is out of toner

Format: `$\{OPENNMS_HOME}/bin/send-event.pl <uei> [host:port] [options]`

The `host:port` defines the hostname and port to send the event; by default, the local host.

Find UEIs and their associated parameters in the `$\{OPENNMS_HOME}/etc/eventd-configuration.xml` file.

> See also xref:events/event-configuration.adoc#ga-events-event-configuration[Event Configuration].

Reviewer 2: ```suggestion
See also xref:events/event-configuration.adoc#ga-events-event-configuration[Event Daemon Configuration].
```

| -V (--version)
| Print version and exit successfully.
|===

> Note that parameter descriptions use the percentage sign (%) as a delimiter to mark a replacement variable.

Reviewer: ```suggestion
Note that parameter descriptions use the percentage sign (%) as a delimiter to mark a replacement variable (see xref:events/event-tokens.adoc#replacement-tokens[Replacement tokens]).
```

> (See xref:events/event-tokens.adoc#replacement-tokens[Replacement tokens].)

Reviewer: ```suggestion
```

| -V (--version)
| Print version and exit successfully.
|===

> Note that parameter descriptions use the percentage sign (%) as a delimiter to mark a replacement variable.

Reviewer 2: Line 115, replacement tokens link does not go to the right place.
Line 117, replace colon with comma after "For example,"

`'testUrl \https://www.opennms.org/ReportSection?filter=FOO_BAR_BAZ%%20eq%%20%%27101%%27'`

==== Example

> Send an event to reload the configuration of Eventd.

Reviewer 2: ```suggestion
Send an event to reload the Eventd configuration.
```
